editor/src/core/layout/layout-utils.ts

```diff
@@ -122,13 +129,39 @@ export function maybeSwitchLayoutProps(
   const newParent = MetadataUtils.findElementByElementPath(currentContextMetadata, newParentPath)
 
   let wasFlexContainer = MetadataUtils.isFlexLayoutedContainer(originalParent)
+  const oldParentMainAxis: 'horizontal' | 'vertical' | null = eitherToMaybe(
+    flatMapEither<string, JSXElementChild, 'horizontal' | 'vertical'>((parentElement) => {
+      if (isJSXElement(parentElement)) {
+        return FlexLayoutHelpers.getMainAxis(right(parentElement.props))
+      } else {
+        return leftEither('parent is not JSXElement')
+      }
+    }, newParent?.element ?? leftEither('no parent provided')),
```

Comment:
This parameter could do with pulling out into a variable to make this expression a little simpler.

Reply:
fixed